plotly/tools.py

```diff
@@ -2509,6 +2561,174 @@ def create_dendrogram(X, orientation="bottom", labels=None,
+            values of the z matrix are annotated. Default = z matrix values.
+        :param (list) fontcolor: List of two colorstrings: [min_text_color,
+            max_text_color] where min_text_color is applied to annotations for
+            heatmap values < (max_value - min_value)/2. If fontcolor is not
+            defined the colors are defined logically as black or white
+            depending on the heatmap's colorscale.
+        :param (bool) showscale: Display colorscale. Default = False
+        :param kwargs: kwargs passed through plotly.graph_objs.Heatmap.
+            These kwargs describe other attributes about the annotated Heatmap
+            trace such as the colorscale. For more information on valid kwargs
+            call help(plotly.graph_objs.Heatmap)
+
+        Example 1: Simple annotated heatmap with default configuration
+        ```
+        import plotly.plotly as py
+        from plotly.tools import FigureFactory as FF
+
+        z = [[0.300000, 0.00000, 0.65, 0.300000], 
+             [1, 0.100005, 0.45, 0.4300],
+             [0.300000, 0.00000, 0.65, 0.300000],
+             [1, 0.100005, 0.45, 0.00000]]
+
+        hm = FF.create_annotated_heatmap(z)
+        py.iplot(hm)
+        ```
+        """
+        # TODO: protected until #282
+        from plotly.graph_objs import graph_objs
+        FigureFactory._validate_annotated_heatmap(z, text)
+        annotations = _AnnotatedHeatmap(z, x, y, text, fontcolor,
+                                        **kwargs).make_annotations()
+
+        if x or y:
+            trace = dict(type='heatmap',
+                         z=z,
+                         x=x,
+                         y=y,
+                         text=hover_text,
+                         showscale=showscale,
+                         **kwargs)
+            layout = dict(annotations=annotations,
+                          xaxis=dict(ticks='', side='top',
+                                     gridcolor='rgb(0, 0, 0)'),
+                          yaxis=dict(ticks='', ticksuffix='  '))
+        else:
+            trace = dict(type='heatmap',
+                         z=z,
+                         text=hover_text,
+                         hoverinfo='text',
+                         showscale=showscale,
+                         **kwargs)
+            layout = dict(annotations=annotations,
+                          xaxis=dict(ticks='', side='top',
+                                     gridcolor='rgb(0, 0, 0)',
+                                     showticklabels=False),
+                          yaxis=dict(ticks='', ticksuffix='  ',
+                                     showticklabels=False))
+
+        data = [trace]
+
+        return graph_objs.Figure(data=data, layout=layout)
+
+    @staticmethod
+    def create_table(text, colorscale=[[0, '#66b2ff'], [.5, '#e6e6e6'],
+                     [1, '#ffffff']], fontcolor=['#000000'],
```

Comment:
@chriddyp I thought these colors looked pretty good for the default table colors but I'm super open to other suggestions